### PR TITLE
chore(wal): allow table to be dropped/renamed while ingesting data via ILP

### PR DESCRIPTION
**Pull request by eugenels**
This PR prevents data loss during ILP ingestion if the table has been renamed in parallel. After renaming, the table will be recreated with the old name if the ILP connection continues to receive data. The same behavior applies if the table is dropped. The new data will be placed in the recreated table. The previous state will be lost.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 101 / 124 (81.45%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/ConcurrentHashMap.java|57|78|73.08%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|15|16|93.75%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|13|13|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryRW.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/seq/TableSequencerAPI.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryRO.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 100 / 126 (79.37%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/ConcurrentHashMap.java|57|78|73.08%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|13|17|76.47%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|14|14|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryRW.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/seq/TableSequencerAPI.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryRO.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 100 / 126 (79.37%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/std/ConcurrentHashMap.java|57|78|73.08%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|13|17|76.47%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|14|14|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryRW.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/seq/TableSequencerAPI.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableNameRegistryRO.java|6|6|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 33 / 38 (86.84%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|13|17|76.47%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|14|14|100.00%|

File: core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpReceiverTest.java [hunk 1]
```diff
@@ -1258,6 +1261,144 @@ public void testWriterInsertNewSymbolsIntoTableWithExistingSymbols() throws Exce
...
+
+    @Test
+    public void testRenameTableSameMeta() throws Exception {
+        Assume.assumeTrue(walEnabled);
+        configOverrideMaxUncommittedRows(2);
+        configOverrideWalSegmentRolloverRowCount(2);
+        FilesFacade filesFacade = new TestFilesFacadeImpl() {
+            @Override
+            public int openRW(LPSZ name, long opts) {
+                if (Chars.endsWith(name, "weather~2" + Files.SEPARATOR + "wal1" + Files.SEPARATOR + "1.lock")) {
```
Comment (ideoma) [inline core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpReceiverTest.java:1357]: Can this `~2` be not hardcoded? TableId or directory format can be changed in the future and this test become unstable.

File: core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpReceiverTest.java [hunk 2]
```diff
@@ -1258,6 +1261,144 @@ public void testWriterInsertNewSymbolsIntoTableWithExistingSymbols() throws Exce
...
+    public void testDropTable() throws Exception {
+        Assume.assumeTrue(walEnabled);
+        configOverrideMaxUncommittedRows(2);
+        configOverrideWalSegmentRolloverRowCount(2);
+        FilesFacade filesFacade = new TestFilesFacadeImpl() {
+            @Override
+            public int openRW(LPSZ name, long opts) {
+                if (Chars.endsWith(name, "weather~2" + Files.SEPARATOR + "wal1" + Files.SEPARATOR + "1.lock")) {
+                    mayDrainWalQueue();
+                    dropWeatherTable();
```
Comment (ideoma) [inline core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpReceiverTest.java:1274]: If I add line 

```java
Os.sleep(1000);
```

data assertion fail. Can you fix or explain why?

Screenshot 2023-01-20 at 14 33 32

File: core/src/main/java/io/questdb/cutlass/line/tcp/TableUpdateDetails.java [hunk 3]
```diff
@@ -241,6 +246,13 @@ private void commit(boolean withLag) throws CommitFailedException {
                 throw CommitFailedException.instance(ex);
             }
         }
+        if (isWal()) {
+            try {
+                engine.verifyTableToken(tableToken);
```
[the remark this answers is not in the document]
Comment (eugenels) [reply core/src/main/java/io/questdb/cutlass/line/tcp/TableUpdateDetails.java]: Fixed in https://github.com/questdb/questdb/pull/2926/commits/0f1dde3cf393ebfbf8b6d2d8a2a7a3b81d51dfd6

File: core/src/main/java/io/questdb/cairo/CairoEngine.java [hunk 4]
```diff
@@ -859,7 +859,7 @@ private void validNameOrThrow(CharSequence tableName) {
         }
     }
 
-    private void verifyTableToken(TableToken tableToken) {
+    public void verifyTableToken(TableToken tableToken) {
```
[the remark this answers is not in the document]
Comment (eugenels) [reply core/src/main/java/io/questdb/cairo/CairoEngine.java]: Fixed in https://github.com/questdb/questdb/pull/2926/commits/0f1dde3cf393ebfbf8b6d2d8a2a7a3b81d51dfd6

File: core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpReceiverTest.java (hunk 1 above)
Comment (eugenels) [reply core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpReceiverTest.java]: Fixed in https://github.com/questdb/questdb/pull/2926/commits/0f1dde3cf393ebfbf8b6d2d8a2a7a3b81d51dfd6

File: core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpReceiverTest.java (hunk 2 above)
Comment (eugenels) [reply core/src/test/java/io/questdb/cutlass/line/tcp/LineTcpReceiverTest.java:1277]: Fixed in https://github.com/questdb/questdb/pull/2926/commits/a044c01937630b933b4e5704a1a949a865c6c09a

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 35 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/TableUpdateDetails.java|10|14|71.43%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpNetworkIOJob.java|6|7|85.71%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|14|14|100.00%|

**Comment by ideoma** [comment]
Please add description

**Review by ideoma — APPROVED** [review]

**Pull request merged**